src/segment/segmentBase.js

```diff
@@ -154,6 +154,10 @@ export const addSidxSegmentsToPlaylist = (playlist, sidx, baseUrl) => {
     number++;
   }
 
+  if (segments.length > 0 && segments[segments.length - 1]) {
```

Comment:
Why are you checking for segments[segments.length - 1]? it must exist if length > 0